test/e2e/tests/confirmations/transactions/gas-fee-tokens-smart-transactions.spec.ts

```diff
@@ -0,0 +1,193 @@
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }: { driver: Driver; localNodes: Anvil }) => {
+        await unlockWallet(driver);
+        await createDappTransaction(driver);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const transactionConfirmation = new TransactionConfirmation(driver);
+        await transactionConfirmation.clickAdvancedDetailsButton();
+        await transactionConfirmation.clickGasFeeTokenPill();
+
+        const gasFeeTokenModal = new GasFeeTokenModal(driver);
+        await gasFeeTokenModal.check_AmountFiat('DAI', '$3.21');
+        await gasFeeTokenModal.check_AmountToken('DAI', '3.21 DAI');
+        await gasFeeTokenModal.check_Balance('DAI', '$10.00');
+
+        await gasFeeTokenModal.check_AmountFiat('USDC', '$1.23');
+        await gasFeeTokenModal.check_AmountToken('USDC', '1.23 USDC');
+        await gasFeeTokenModal.check_Balance('USDC', '$5.00');
+        await gasFeeTokenModal.clickToken('USDC');
+
+        await transactionConfirmation.check_gasFeeSymbol('USDC');
+        await transactionConfirmation.check_gasFeeFiat('$1.23');
+        await transactionConfirmation.check_gasFee('1.23');
+        await transactionConfirmation.check_gasFeeTokenFee('$0.43');
+        await transactionConfirmation.clickFooterConfirmButton();
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        const homepage = new HomePage(driver);
+        await homepage.goToActivityList();
+
+        const activityListPage = new ActivityListPage(driver);
+        await activityListPage.check_confirmedTxNumberDisplayedInActivity(2);
+      },
+    );
+  });
+
+  it('fails two transactions if error', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder({ inputChainId: CHAIN_IDS.MAINNET })
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesControllerSmartTransactionsOptedIn()
+          .withNetworkControllerOnMainnet()
+          .build(),
+        localNodeOptions: {
+          hardfork: 'london',
+        },
+        testSpecificMock: (mockServer: MockttpServer) => {
+          mockSimulationResponse(mockServer);
+          mockSmartTransactionBatchRequests(mockServer, {
+            transactionHashes: [TRANSACTION_HASH, TRANSACTION_HASH_2],
+            error: true,
+          });
+        },
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }: { driver: Driver; localNodes: Anvil }) => {
+        await unlockWallet(driver);
+        await createDappTransaction(driver);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const transactionConfirmation = new TransactionConfirmation(driver);
+        await transactionConfirmation.clickGasFeeTokenPill();
+
+        const gasFeeTokenModal = new GasFeeTokenModal(driver);
+        await gasFeeTokenModal.clickToken('USDC');
```

Comment:
it seems that there's a race condition, where in FF, the selected token is ETH, instead of USDC :thinking: 

https://github.com/user-attachments/assets/34f49f1f-c0f2-4e66-8751-7b8df1aefab0